src/component/child/index.js

```diff
@@ -76,6 +80,23 @@ export class ChildComponent extends BaseComponent {
         });
     }
 
+    hasValidParentDomain() {
+        const parentDomain = this.getParentDomain();
+        if (typeof this.component.allowedParentDomains === 'string') {
```

Comment:
You can use `matchDomain()` for arrays and regexes too, so no need to code a loop here at all